File: 1000/Filling_Shapes.cpp
/*
Author: passa
Task #: 1182A
Task Name: Filling Shapes
Difficulty: 1000
Language: C++ 
*/

#include <bits/stdc++.h>
using namespace std;

int n;
long answer;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(NULL);

    cin >> n;

    if (n % 2 != 0)
        cout << 0 << endl;

    else {
        answer = pow(2, n/2);
        cout << answer << endl;
    }

    return 0;
}

File: 1000/Filling_Shapes.py
"""
Author: passa
Task #: 1182A
Task Name: Filling Shapes
Difficulty: 1000
Language: Python
"""

def solution(n):
    if n % 2 != 0:
        return 0
  
    return 2 ** int(n/2)

n = int(input())

print(solution(n))